### PR TITLE
Updated datasource version parsing

### LINKED ISSUE #4582: Funcotator - review and finalize data sources release numbers / versioning
Need to look at how to version the releases of data sources and individual data sources.

This is exemplified with the dbSNP `common` vs dbSNP `All` problem.

### LINKED ISSUE #4692: Funcotator: don't parse README.txt, use a manifest instead

### PR DESCRIPTION
**Pull request by jonn-smith**
Now looks at manifest file, not readme.
Now supports version decorators (e.g. `somatic`) after version numbers in manifest file.
Added in a unit test to check the version regex.

Fixes #4582 
Fixes #4692

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/DataSourceUtils.java [hunk 1]
```diff
@@ -461,12 +462,14 @@ private static Properties readConfigFileProperties(final Path configFilePath) {
      * user can create their own data sources directory, which may not contain the metadata we seek.
      *
      * NOTE: The README file in a Data Sources directory is assumed to have the following properties:
```
Comment (LeeTL1220) [inline src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/DataSourceUtils.java]: Should this now read:  `The MANIFEST file in a Data ....`?

File: src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/DataSourceUtils.java [hunk 2]
```diff
@@ -63,19 +65,18 @@ private DataSourceUtils() {}
...
-    static final int MIN_DAY_RELEASED         = 15;
+    static final int MIN_DAY_RELEASED         = 29;
 
     //==================================================================================================================
     // Public Static Members:
 
     /** The minimum version of the data sources required for funcotator to run.  */
     public static final String CURRENT_MINIMUM_DATA_SOURCE_VERSION         = String.format("v%d.%d.%d%02d%02d", MIN_MAJOR_VERSION_NUMBER, MIN_MINOR_VERSION_NUMBER, MIN_YEAR_RELEASED, MIN_MONTH_RELEASED, MIN_DAY_RELEASED);
-    public static final String README_FILE_NAME                            = "README.txt";
+    public static final String MANIFEST_FILE_NAME                          = "MANIFEST.txt";
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/DataSourceUtils.java]: Did you just rename README.txt to MANIFEST.txt, or is this a more substantial change? Ie., is `MANIFEST.txt` now a structured format, or is it still basically a README with another name?

File: src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/DataSourceUtils.java (hunk 1 above)
Comment (jonn-smith) [reply src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/DataSourceUtils.java]: Yes.  Fixed!

File: src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/DataSourceUtils.java (hunk 2 above)
Comment (jonn-smith) [reply src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/DataSourceUtils.java]: @droazen I added a `MANIFEST.txt` file to the packages.  `MANIFEST.txt` contains the version and provenance information for each data source package, as well as a new decorator/information field to distinguish somatic and germline files.  This version information is duplicated in the readme, but the readme is not parsed.

`MANIFEST.txt` a plain-text formatted file with 1 field per line.  The formatting is loosely enforced with the regex in `DataSourceUtils` and is of the syntax:

<FIELD>: <VALUE>

Comment (droazen) [reply src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/DataSourceUtils.java]: Ok, sounds good @jonn-smith, thanks for the clarification.

**Pull request merged**